src/main/java/org/broadinstitute/hellbender/tools/spark/sv/FindBreakpointEvidenceSpark.java

```diff
@@ -181,36 +197,78 @@ protected void runTool( final JavaSparkContext ctx ) {
-        final Map<Integer, String> intervalDispositions;
+        final List<AssemblyResults> assemblyResultsList;
         if ( intervalOnlyAssembly ) {
-            intervalDispositions = new HashMap<>();
+            assemblyResultsList = new ArrayList<>();
         } else {
-            intervalDispositions = addAssemblyQNames(params, ctx, kmersToIgnoreFile, qNamesMultiMap, intervals.size(),
+            assemblyResultsList = addAssemblyQNames(params, ctx, kmersToIgnoreFile, qNamesMultiMap, intervals.size(),
                     allPrimaryLines, locations, pipelineOptions);
         }
 
         // write a FASTQ file for each interval
-        final String outDir = outputDir;
-        final int maxFastqSize = maxFASTQSize;
-        final boolean includeMapLoc = includeMappingLocation;
-        intervalDispositions.putAll(
-                generateFastqs(ctx, qNamesMultiMap, allPrimaryLines, intervals.size(), includeMapLoc,
-                                intervalAndFastqBytes -> writeFastq(intervalAndFastqBytes, outDir, maxFastqSize)));
+        final FastqWriter fastqWriter =
+                new FastqWriter(locations.alignerIndexFile, maxFASTQSize, fastqDir, gfaDir);
+        assemblyResultsList.addAll(
+                generateFastqs(ctx, qNamesMultiMap, allPrimaryLines, intervals.size(),
+                                includeMappingLocation, fastqDir != null, fastqWriter));
+
+        assemblyResultsList.sort( (result1, result2) ->
+                Integer.compare(result1.getAssemblyId(), result2.getAssemblyId()) );
+
+        // write the output file
+        writeSAMFile(outputSAM, pipelineOptions, header, assemblyResultsList);
 
         // record the intervals
         if ( locations.intervalFile != null ) {
-            writeIntervalFile(locations.intervalFile, pipelineOptions, header, intervals, intervalDispositions);
+            writeIntervalFile(locations.intervalFile, pipelineOptions, header, intervals, assemblyResultsList);
         }
 
         log("Wrote FASTQs for assembly.");
     }
 
+    /** write a SAM file containing records for each aligned contig */
+    private static void writeSAMFile( final String samFile, final PipelineOptions pipelineOptions,
+                                      final SAMFileHeader header, final List<AssemblyResults> assemblyResultsList ) {
+        try ( final OutputStream os = BucketUtils.createFile(samFile, pipelineOptions) ) {
+            final SAMTextWriter writer = new SAMTextWriter(os);
+            writer.setSortOrder(SAMFileHeader.SortOrder.queryname, true);
+            writer.setHeader(header);
+            final List<String> refNames =
+                header.getSequenceDictionary().getSequences().stream()
+                        .map(SAMSequenceRecord::getSequenceName).collect(Collectors.toList());
+            for ( final AssemblyResults assemblyResults : assemblyResultsList ) {
+                if ( assemblyResults.getErrorMessage() != null ) continue;
```

Comment:
There are a couple of reasons that intervals don't get assembled (too big is one, too few kmers is another), and the idea of the AssemblyResults class is to present either the assembly, or the reason why there isn't one.  Though we can't write contigs that we didn't generate, the errors aren't silent.  They'll be output in the intervalFile.